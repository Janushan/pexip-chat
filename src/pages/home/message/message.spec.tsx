import React from "react";
import { shallow } from "enzyme";
import { Button } from "@material-ui/core";

import Message from ".";

describe("<Message/>", () => {
  const Props = {
    id: "256b3a6b-4639-48ec-96b5-be1943be7617",
    name: "Janu Shan",
    timeStamp: "1591307010490",
    content: "Hi there",
    onEditClick: jest.fn(),
    onDeleteClick: jest.fn(),
    status: false,
    action: "",
    showActions: true,
  };

  it("renders message WITH action buttons - edit/delete", () => {
    const props = {
      ...Props,
      showActions: true,
    };
    const wrapper = shallow(<Message {...props} />).debug();

    expect(wrapper).toMatchSnapshot();
  });

  it("renders message WITHOUT action buttons - edit/delete", () => {
    const props = {
      ...Props,
      showActions: false,
    };
    const wrapper = shallow(<Message {...props} />).debug();

    expect(wrapper).toMatchSnapshot();
  });

  it("it should call onEditClick when save button is pressed", () => {
    const props = {
      ...Props,
      showActions: true,
    };
    const wrapper = shallow(<Message {...props} />);
    wrapper.find(Button).at(0).simulate("click");
    wrapper.find(Button).at(0).simulate("click");
    expect(Props.onEditClick).toHaveBeenCalled();
  });

  it("it should call onDeleteClick when delete button is pressed", () => {
    const props = {
      ...Props,
      showActions: true,
    };
    const wrapper = shallow(<Message {...props} />);
    wrapper.find(Button).at(1).simulate("click");
    expect(Props.onDeleteClick).toHaveBeenCalled();
  });
});
