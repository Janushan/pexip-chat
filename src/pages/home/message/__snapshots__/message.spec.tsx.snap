// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<Message/> renders message WITH action buttons - edit/delete 1`] = `
"<WithStyles(ForwardRef(Grid)) container={true} className=\\"message\\" alignItems=\\"center\\">
  <WithStyles(ForwardRef(Grid)) container={true} alignItems=\\"center\\">
    <WithStyles(ForwardRef(Grid))>
      <WithStyles(ForwardRef(Grid)) container={true} direction=\\"row\\" alignItems=\\"center\\">
        <WithStyles(ForwardRef(Typography)) variant=\\"h5\\" className=\\"name\\">
          Janu Shan
        </WithStyles(ForwardRef(Typography))>
        <WithStyles(ForwardRef(Typography)) variant=\\"body1\\" className=\\"timeStamp\\">
          Invalid date
        </WithStyles(ForwardRef(Typography))>
      </WithStyles(ForwardRef(Grid))>
      <Linkify componentDecorator={[Function]} hrefDecorator={[Function]} matchDecorator={[Function]} textDecorator={[Function]}>
        <WithStyles(ForwardRef(Grid)) container={true} direction=\\"row\\" alignItems=\\"center\\">
          <WithStyles(ForwardRef(Grid)) container={true}>
            <WithStyles(ForwardRef(Typography)) variant=\\"body1\\" className=\\"content\\">
              Hi there
            </WithStyles(ForwardRef(Typography))>
          </WithStyles(ForwardRef(Grid))>
          <WithStyles(ForwardRef(Button)) variant=\\"outlined\\" className=\\"button\\" onClick={[Function: onClick]}>
            Edit
          </WithStyles(ForwardRef(Button))>
          <WithStyles(ForwardRef(Button)) variant=\\"outlined\\" className=\\"deleteButton\\" onClick={[Function: onClick]}>
            Delete
          </WithStyles(ForwardRef(Button))>
        </WithStyles(ForwardRef(Grid))>
      </Linkify>
    </WithStyles(ForwardRef(Grid))>
  </WithStyles(ForwardRef(Grid))>
</WithStyles(ForwardRef(Grid))>"
`;

exports[`<Message/> renders message WITHOUT action buttons - edit/delete 1`] = `
"<WithStyles(ForwardRef(Grid)) container={true} className=\\"message\\" alignItems=\\"center\\">
  <WithStyles(ForwardRef(Grid)) container={true} alignItems=\\"center\\">
    <WithStyles(ForwardRef(Grid))>
      <WithStyles(ForwardRef(Grid)) container={true} direction=\\"row\\" alignItems=\\"center\\">
        <WithStyles(ForwardRef(Typography)) variant=\\"h5\\" className=\\"name\\">
          Janu Shan
        </WithStyles(ForwardRef(Typography))>
        <WithStyles(ForwardRef(Typography)) variant=\\"body1\\" className=\\"timeStamp\\">
          Invalid date
        </WithStyles(ForwardRef(Typography))>
      </WithStyles(ForwardRef(Grid))>
      <Linkify componentDecorator={[Function]} hrefDecorator={[Function]} matchDecorator={[Function]} textDecorator={[Function]}>
        <WithStyles(ForwardRef(Typography)) variant=\\"body1\\" className=\\"content\\">
          Hi there
        </WithStyles(ForwardRef(Typography))>
      </Linkify>
    </WithStyles(ForwardRef(Grid))>
  </WithStyles(ForwardRef(Grid))>
</WithStyles(ForwardRef(Grid))>"
`;
